### PR TITLE
MF-811 Added MQTT troubleshooting section

### LINKED ISSUE #811: MQTT  fails when numerous clients with bad creds try to publish
**BUG REPORT**

1. What were you trying to achieve?
benchmark the mqtt

2. What are the expected results?
mqtt working properly

3. What are the received results?
mqtt fails

```
00000000\",\"redis_git_dirty\":\"0\",\"redis_build_id\":\"cbfa5816ce932578\",\"redis_mode\":\"standalone\",\"os\":\"Linux 4.15.0-54-generic x86_64\",\"arch_bits\":\
mainflux-mqtt   | <--- Last few GCs --->
mainflux-mqtt   | 
mainflux-mqtt   | [1:0x561dc049d800]    13937 ms: Mark-sweep 972.3 (1029.9) -> 972.3 (994.4) MB, 88.0 / 0.0 ms  (average mu = 0.863, current mu = 0.000) last resort GC in old space requested
mainflux-mqtt   | [1:0x561dc049d800]    14025 ms: Mark-sweep 972.3 (994.4) -> 972.3 (993.9) MB, 88.0 / 0.0 ms  (average mu = 0.751, current mu = 0.000) last resort GC in old space requested
mainflux-mqtt   | 
mainflux-mqtt   | 
mainflux-mqtt   | <--- JS stacktrace --->
mainflux-mqtt   | 
mainflux-mqtt   | ==== JS stack trace =========================================
mainflux-mqtt   | 
mainflux-mqtt   |     0: ExitFrame [pc: 0x5234825be1d]
mainflux-mqtt   |     1: StubFrame [pc: 0x52348214822]
mainflux-mqtt   |     2: StubFrame [pc: 0x5234853598e]
mainflux-mqtt   | Security context: 0x286e0c7f2229 <JSObject>
mainflux-mqtt   |     3: flatstr(aka flatstr) [0x2cf147b10579] [/node_modules/flatstr/index.js:11] [bytecode=0x8223e86ef71 offset=3](this=0x3c1799e826f1 <undefined>,s=0x08223e806581 <Very long string[637855849]>)
mainflux-mqtt   |     4: pinoWrite [0x3364a67ffdb9] [/node_modules/pino/pino.js:225] [bytecode=0x382f8962f919 ...
mainflux-mqtt   | 
mainflux-mqtt   | FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
mainflux-mqtt exited with code 137

```

4. What are the steps to reproduce the issue?
```
go get github.com/krylovsk/mqtt-benchmark 
mqtt-benchmark --broker tcp://xxx.xxx.xxx.xxx:1883 --count 1000 --size 100 --clients 1000 --qos 2 --format text --password="$KEY" --username="$THING" --topic="channels/$CH/messages/test"
```

5. In what environment did you encounter the issue?
client pc ubuntu 18.04
mainflux server DO machine ubuntu 18.04 RAM 2GB

6. Additional information you deem important:

### PR DESCRIPTION
**Pull request by chombium**
### What does this do?

This PR improves the developer guide, by adding a section for troubleshooting fixing problems related to the Aedes MQTT adapter. It also adds a suggestion how to fix the problem noticed in #811. 

### Which issue(s) does this PR fix/relate to?
Fixes #811 

### List any changes that modify/break current functionality
None.

### Have you included tests for your changes?

No. This PR enhances the dev guide

### Did you document any new/modified functionality?
Yes.

### Notes

If we see that we have to add troubleshooting sections for some of the other services it might be also good if we think about creating a new troubleshooting guide, so that we don't put to many things in the dev guide.

### CODE REVIEW

**Review by drasko — DISMISSED** [review]
LGTM

File: docs/dev-guide.md [hunk 1]
```diff
@@ -133,6 +133,18 @@ After that, the MQTT Adapter can be started from top directory (as it needs to f
...
+
+Up until now it has been noticed that in case of high load, big messages and many client it can happen that the MQTT client fails with the following error:
+```
+mainflux-mqtt   | FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
+mainflux-mqtt exited with code 137
+```
+The problem is the default allowed memory in node (V8). [V8 gives the user 1.7GB per default](https://medium.com/tomincode/increasing-nodes-memory-337dfb1a60dd). To fix the problem you should add the following environment variable `NODE_OPTIONS:--max-old-space-size=SPACE_IN_MB` in the [environment section](https://github.com/mainflux/mainflux/blob/master/docker/aedes.yml#L31) of the aedes.yml configuration. To find the right value for the `--max-old-space-size` parameter you'll have to experiment a bit depending on your needs.
+
+The Mainflux MQTT service uses the [Aedes MQTT Broker](https://github.com/mcollina/aedes) for implementation of the MQTT releated things. Therefore, for some quesions or problems you can also check out the Aedes's documentation or reach out it's contributors.
```
Comment (dborovcanin) [inline docs/dev-guide.md]: Please fix typos: `fequency`, `releated`, `quesions`, `it's contributors`.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: docs/dev-guide.md (hunk 1 above)
Comment (chombium) [reply docs/dev-guide.md]: @dusanb94 Shame on me! I guess I was to tired when I added this :sleeping: :worried: . Now I was more careful and the spell checker was turned on :wink:

**Review by drasko — APPROVED** [review]
LGTM

File: docs/dev-guide.md (hunk 1 above)
Comment (dborovcanin) [reply docs/dev-guide.md]: Thanks, @chombium!

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**